src/main/java/core/basesyntax/strategy/impl/ApplyPurchase.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.FruitStoreException;
+import core.basesyntax.model.FruitNegotiation;
+import core.basesyntax.strategy.UnaryOperation;
+
+public class ApplyPurchase implements UnaryOperation {
+    private FruitNegotiation myFruit;
+
+    public ApplyPurchase(FruitNegotiation fruit) {
+        this.myFruit = fruit;
+    }
+
+    @Override
+    public void apply() {
+        if (Storage.storage.containsKey(myFruit.getFruit())) {
+            if (Storage.storage.get(myFruit.getFruit()).intValue() >= myFruit.getQuantity()) {
+                Storage.storage.put(myFruit.getFruit(),
+                        Storage.storage.get(myFruit.getFruit()).intValue() - myFruit.getQuantity());
+            } else {
+                throw new FruitStoreException("Can't selling " + myFruit.getFruit()
+                        + " the quantity of which is less on the balance than in query");
+            }
+        } else {
+            throw new FruitStoreException("Can't selling " + myFruit.getFruit()
+                    + " fruit that don't getting on balance before");
+        }
```

Comment:
let's not overcomplicate the `if`s, it is almost always better to just do the checks like:
```
if (something wrong) {
    throw exception;
} // no else because it is redundant
// business logic
```
thus it is more clear to understand

Reply:
also check similar sitiations

Reply:
refactored